byron/ledger/executable-spec/src/Ledger/Delegation.hs

```diff
@@ -575,3 +589,124 @@ initialEnvFromGenesisKeys ngk chainLength =
     <*> epochGen 0 10
     <*> slotGen 0 100
     <*> CoreGen.k chainLength (chainLength `div` 10)
+
+--------------------------------------------------------------------------------
+-- Shared support functions for Delegation Properties
+--------------------------------------------------------------------------------
+
+delegatorDelegate :: DCert -> (VKeyGenesis, VKey)
+delegatorDelegate = delegator &&& delegate
+
+ratioInt :: Int -> Int -> Double
+ratioInt x y
+  = fromIntegral x / fromIntegral y
+
+-- | Filters the list with the predicate and returns
+-- the ratio of filtered to original list lengths.
+lenRatio :: ([a] -> [b]) -> [a] -> Double
+lenRatio p xs
+  = ratioInt (length (p xs))
+             (length xs)
+
+maxInt :: [Int] -> Int
+maxInt [] = 0
+maxInt xs = List.maximum xs
+
+-- | True if the tuple count (snd item) is >= 2
+multiple :: (a,Int) -> Bool
+multiple = (2 <=) . snd
+
+-- | Count the number of delegates in the given certificates
+delegateCounts :: [DCert] -> [(VKey, Int)]
+delegateCounts certs
+  = fmap delegatorDelegate certs
+    -- Remove duplicated elements, since we're not
```

Comment:
BTW, I think we can expand the line length of the comments (if you're using Emacs you can reformat this with `M-q`)